From 071dabcd14ecb1c019aece0db834b6cd6e32bae3 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Tue, 29 Jul 2025 14:45:50 +0200
Subject: [PATCH] Revert "drm/panel: otm8009a: Use runtime pm to avoid
 excessive unprepare / prepare"

This reverts commit 68fb469bf983899deec3b867c4eadd1c28575928.
This property was not added in the binding & is no longer needed.

Change-Id: Id24f43d841d1ee6c8a9d3a143c3b91a4e7ec2a26
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/479217
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../gpu/drm/panel/panel-orisetech-otm8009a.c  | 73 ++++---------------
 1 file changed, 16 insertions(+), 57 deletions(-)

--- a/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
+++ b/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
@@ -10,7 +10,6 @@
 #include <linux/delay.h>
 #include <linux/gpio/consumer.h>
 #include <linux/module.h>
-#include <linux/pm_runtime.h>
 #include <linux/regulator/consumer.h>
 
 #include <video/mipi_display.h>
@@ -294,16 +293,16 @@ static int otm8009a_disable(struct drm_p
 static int otm8009a_unprepare(struct drm_panel *panel)
 {
 	struct otm8009a *ctx = panel_to_otm8009a(panel);
-	int ret;
 
 	if (!ctx->prepared)
 		return 0;
 
-	pm_runtime_set_autosuspend_delay(ctx->dev, 1000);
-	pm_runtime_mark_last_busy(panel->dev);
-	ret = pm_runtime_put_autosuspend(panel->dev);
-	if (ret < 0)
-		return ret;
+	if (ctx->reset_gpio) {
+		gpiod_set_value_cansleep(ctx->reset_gpio, 1);
+		msleep(20);
+	}
+
+	regulator_disable(ctx->supply);
 
 	ctx->prepared = false;
 
@@ -318,12 +317,20 @@ static int otm8009a_prepare(struct drm_p
 	if (ctx->prepared)
 		return 0;
 
-	ret = pm_runtime_get_sync(panel->dev);
+	ret = regulator_enable(ctx->supply);
 	if (ret < 0) {
-		pm_runtime_put_autosuspend(panel->dev);
+		dev_err(panel->dev, "failed to enable supply: %d\n", ret);
 		return ret;
 	}
 
+	if (ctx->reset_gpio) {
+		gpiod_set_value_cansleep(ctx->reset_gpio, 0);
+		gpiod_set_value_cansleep(ctx->reset_gpio, 1);
+		msleep(20);
+		gpiod_set_value_cansleep(ctx->reset_gpio, 0);
+		msleep(100);
+	}
+
 	ret = otm8009a_init_sequence(ctx);
 	if (ret)
 		return ret;
@@ -467,18 +474,9 @@ static int otm8009a_probe(struct mipi_ds
 			  MIPI_DSI_MODE_LPM | MIPI_DSI_CLOCK_NON_CONTINUOUS |
 			  MIPI_DSI_MODE_NO_EOT_PACKET;
 
-	pm_runtime_enable(ctx->dev);
-	/* set delay to 60s to keep alive the panel to wait the splash screen */
-	pm_runtime_set_autosuspend_delay(ctx->dev, 60000);
-	pm_runtime_use_autosuspend(ctx->dev);
-
 	drm_panel_init(&ctx->panel, dev, &otm8009a_drm_funcs,
 		       DRM_MODE_CONNECTOR_DSI);
 
-	pm_runtime_get_sync(ctx->dev);
-	pm_runtime_mark_last_busy(ctx->dev);
-	pm_runtime_put_autosuspend(ctx->dev);
-
 	ctx->bl_dev = devm_backlight_device_register(dev, dev_name(dev),
 						     dev, ctx,
 						     &otm8009a_backlight_ops,
@@ -512,46 +510,8 @@ static void otm8009a_remove(struct mipi_
 
 	mipi_dsi_detach(dsi);
 	drm_panel_remove(&ctx->panel);
-
-	pm_runtime_dont_use_autosuspend(ctx->dev);
-	pm_runtime_disable(ctx->dev);
-}
-
-static __maybe_unused int orisetech_otm8009a_suspend(struct device *dev)
-{
-	struct otm8009a *ctx = dev_get_drvdata(dev);
-
-	gpiod_set_value_cansleep(ctx->reset_gpio, 1);
-	msleep(20);
-
-	regulator_disable(ctx->supply);
-
-	return 0;
 }
 
-static __maybe_unused int orisetech_otm8009a_resume(struct device *dev)
-{
-	struct otm8009a *ctx = dev_get_drvdata(dev);
-	int ret;
-
-	ret = regulator_enable(ctx->supply);
-	if (ret < 0) {
-		dev_err(ctx->dev, "failed to enable supply: %d\n", ret);
-		return ret;
-	}
-
-	gpiod_set_value_cansleep(ctx->reset_gpio, 0);
-	msleep(100);
-
-	return 0;
-}
-
-static const struct dev_pm_ops orisetech_otm8009a_pm_ops = {
-	SET_RUNTIME_PM_OPS(orisetech_otm8009a_suspend, orisetech_otm8009a_resume, NULL)
-	SET_SYSTEM_SLEEP_PM_OPS(pm_runtime_force_suspend,
-				pm_runtime_force_resume)
-};
-
 static const struct of_device_id orisetech_otm8009a_of_match[] = {
 	{ .compatible = "orisetech,otm8009a" },
 	{ }
@@ -564,7 +524,6 @@ static struct mipi_dsi_driver orisetech_
 	.driver = {
 		.name = "panel-orisetech-otm8009a",
 		.of_match_table = orisetech_otm8009a_of_match,
-		.pm = &orisetech_otm8009a_pm_ops,
 	},
 };
 module_mipi_dsi_driver(orisetech_otm8009a_driver);
